[feat] 예약 페이지 구현

khanz0613 opened the pull request:
## 🔖 PR 유형
- [x] ✨ 기능 추가
- [ ] 🐛 버그 수정
- [x] ♻️ 리팩토링
- [ ] 🧪 테스트 코드 추가
- [ ] 📄 문서 수정
- [ ] 기타

## 📌 개요
 팬미팅 예약 시스템의 전체 플로우를 구현했습니다.

🔧 작업 내용
- 팬미팅 리스트 페이지 (FanMeetingPage.vue) 구현:
  - 인플루언서 목록 표시 (태요미네, 여단오, 침착맨 등 8개)
  - 검색 기능 UI 구현
  - 각 인플루언서별 상세 페이지 연결

- 팬미팅 상세 페이지 (FanMeetingDetailPage.vue) 구현:
  - 팬미팅 정보 상세 표시
  - 예매하기 버튼으로 좌석 선택 페이지 연결

- 좌석 선택 페이지 (SeatSelectPage.vue) 구현:
  - 스테이지 중심의 좌석 배치도
  - 좌석 상태별 색상 구분 (선택가능/선택됨/선택불가)
  - 선택된 좌석 정보 실시간 표시
  - 예매하기 버튼으로 결제 페이지 연결

- 결제 페이지 (FanMeetingPaymentPage.vue) 구현:
  - 주문 정보 요약 (팬미팅, 장소, 일시, 좌석, 가격)
  - 결제 방법 선택 (KB페이, 카카오페이, 네이버페이)
  - 약관 동의 체크박스 (임의로 추가)
  - 결제 처리 로딩 상태 및 완료 페이지 연결

- 예약 관련 페이지들을 src/pages/reservation/ 폴더로 구조화

## ♻️ 리팩토링

  - 디자인 시스템 가이드라인 적용:
    - 하드코딩된 컬러를 커스텀 Tailwind 클래스로 변경
    - 인라인 스타일 제거 및 표준 클래스 적용
    - 컨테이너 사이즈 가이드라인 준수
    - 코드 일관성 및 유지보수성 향상

## ✅ 체크리스트
예약 페이지
<img width="428" height="926" alt="image" src="https://github.com/user-attachments/assets/65ce1dfe-978f-499e-85fd-3928119c557f" />

예약 상세 페이지
<img width="426" height="928" alt="image" src="https://github.com/user-attachments/assets/d7a9ea13-5055-4aa1-9f16-f6bfb320219e" />

좌석 선택 페이지(선택 전)
<img width="428" height="929" alt="image" src="https://github.com/user-attachments/assets/fd245fd1-f5e7-4b62-84fa-8a3ee6d0a642" />

좌석 선택 패에지(선택 후)
<img width="427" height="928" alt="image" src="https://github.com/user-attachments/assets/30625ca2-b699-4a8c-94cb-efa6feb5058a" />

결제 페이지(동의 전)
<img width="425" height="930" alt="image" src="https://github.com/user-attachments/assets/5542fa83-e515-42b4-a8e1-fd3fd8aa6026" />

결제 페이지(동의 후)
<img width="427" height="927" alt="image" src="https://github.com/user-attachments/assets/9d81c226-9f9e-4653-aafa-4f771b517e18" />


## 📝 기타 참고 사항
  - 현재는 더미 데이터로 구현되어 있으며, 추후 백엔드 API 연동 예정
  - 실제 결제 기능은 임시 시뮬레이션으로 구현 (2초 딜레이)
  - 좌석 상태 관리 로직 추후 서버 연동 시 개선 예정
  - 결제 완료 페이지 올라오는 대로 연결 및 티켓 보기 버튼 추가 예정
 
## 📎 관련 이슈
Close #18

khanz0613 committed:
[feat] 팬미팅 예약 페이지 기본 구현

 - 팬미팅 리스트 페이지(FanMeetingPage) 인플루언서 목록 및 검색 UI 구현
  - 팬미팅 상세 페이지(FanMeetingDetailPage) 상세 정보 및 예매 버튼 구현
  - 좌석 선택 페이지(SeatSelectPage) 좌석맵 및 선택 기능 구현
  - 결제 페이지(FanMeetingPaymentPage) 주문 정보 및 결제 방법 선택 구현

khanz0613 committed:
[feat] 팬미팅 예약 페이지 기본 구현

 - 팬미팅 리스트 페이지(FanMeetingPage) 인플루언서 목록 및 검색 UI 구현
  - 팬미팅 상세 페이지(FanMeetingDetailPage) 상세 정보 및 예매 버튼 구현
  - 좌석 선택 페이지(SeatSelectPage) 좌석맵 및 선택 기능 구현
  - 결제 페이지(FanMeetingPaymentPage) 주문 정보 및 결제 방법 선택 구현

khanz0613 committed:
[feat] 팬미팅 예약 페이지 기본 구현

 - 팬미팅 리스트 페이지(FanMeetingPage) 인플루언서 목록 및 검색 UI 구현
  - 팬미팅 상세 페이지(FanMeetingDetailPage) 상세 정보 및 예매 버튼 구현
  - 좌석 선택 페이지(SeatSelectPage) 좌석맵 및 선택 기능 구현
  - 결제 페이지(FanMeetingPaymentPage) 주문 정보 및 결제 방법 선택 구현

khanz0613 committed:
[feat] 팬미팅 예약 페이지 기본 구현

 - 팬미팅 리스트 페이지(FanMeetingPage) 인플루언서 목록 및 검색 UI 구현
  - 팬미팅 상세 페이지(FanMeetingDetailPage) 상세 정보 및 예매 버튼 구현
  - 좌석 선택 페이지(SeatSelectPage) 좌석맵 및 선택 기능 구현
  - 결제 페이지(FanMeetingPaymentPage) 주문 정보 및 결제 방법 선택 구현

khanz0613 committed:
[feat] 결제 완료 페이지 신규 추가

- PaymentSuccessPage.vue 컴포넌트 생성
- 결제 완료 후 안내 UI 구현

khanz0613 committed:
[feat] 팬미팅 예약 관련 라우트 추가

  - 팬미팅 상세(/fan-meeting/:id), 좌석 선택(/seat-select/:id), 결제(/payment/:id) 라우트 설정
  - 결제 완료 페이지(/payment-success) 라우트 추가

khanz0613 committed:
[feat] 팬미팅 예약 관련 라우트 추가

  - 팬미팅 상세(/fan-meeting/:id), 좌석 선택(/seat-select/:id), 결제(/payment/:id) 라우트 설정
  - 결제 완료 페이지(/payment-success) 라우트 추가

khanz0613 committed:
[feat] 팬미팅 예약 관련 라우트 추가

  - 팬미팅 상세(/fan-meeting/:id), 좌석 선택(/seat-select/:id), 결제(/payment/:id) 라우트 설정
  - 결제 완료 페이지(/payment-success) 라우트 추가

khanz0613 committed:
[feat] 팬미팅 예약 관련 라우트 추가

  - 팬미팅 상세(/fan-meeting/:id), 좌석 선택(/seat-select/:id), 결제(/payment/:id) 라우트 설정
  - 결제 완료 페이지(/payment-success) 라우트 추가

khanz0613 committed:
[feat] 팬미팅 예약 시스템 UI/UX 개선 및 검색 기능 구현

  - 전역 네비게이션 바 조건부 렌더링 (상세/좌석선택/결제/완료 페이지에서 숨김)
  - 팬미팅 상세 페이지 레이아웃 재구성 (뒤로가기 버튼, 인플루언서 프로필, 포스터 중심)
  - 예약 페이지 검색 기능 활성화 (인플루언서 이름/위치 기반 실시간 필터링)
  - 검색창 UI 개선 (돋보기 아이콘 우측 배치, 위치 텍스트 색상 조정)
  - 아이콘 색상 통일 (#808080) 및 버튼 스타일 일관성 개선
  - 네비게이션 바 제거에 따른 하단 버튼 위치 조정 (bottom-20 → bottom-0)

khanz0613 committed:
[feat] 팬미팅 예약 시스템 UI/UX 개선 및 검색 기능 구현

  - 전역 네비게이션 바 조건부 렌더링 (상세/좌석선택/결제/완료 페이지에서 숨김)
  - 팬미팅 상세 페이지 레이아웃 재구성 (뒤로가기 버튼, 인플루언서 프로필, 포스터 중심)
  - 예약 페이지 검색 기능 활성화 (인플루언서 이름/위치 기반 실시간 필터링)
  - 검색창 UI 개선 (돋보기 아이콘 우측 배치, 위치 텍스트 색상 조정)
  - 아이콘 색상 통일 (#808080) 및 버튼 스타일 일관성 개선
  - 네비게이션 바 제거에 따른 하단 버튼 위치 조정 (bottom-20 → bottom-0)

khanz0613 committed:
[feat] 팬미팅 예약 시스템 UI/UX 개선 및 검색 기능 구현

  - 전역 네비게이션 바 조건부 렌더링 (상세/좌석선택/결제/완료 페이지에서 숨김)
  - 팬미팅 상세 페이지 레이아웃 재구성 (뒤로가기 버튼, 인플루언서 프로필, 포스터 중심)
  - 예약 페이지 검색 기능 활성화 (인플루언서 이름/위치 기반 실시간 필터링)
  - 검색창 UI 개선 (돋보기 아이콘 우측 배치, 위치 텍스트 색상 조정)
  - 아이콘 색상 통일 (#808080) 및 버튼 스타일 일관성 개선
  - 네비게이션 바 제거에 따른 하단 버튼 위치 조정 (bottom-20 → bottom-0)

khanz0613 committed:
[feat] 팬미팅 예약 시스템 UI/UX 개선 및 검색 기능 구현

  - 전역 네비게이션 바 조건부 렌더링 (상세/좌석선택/결제/완료 페이지에서 숨김)
  - 팬미팅 상세 페이지 레이아웃 재구성 (뒤로가기 버튼, 인플루언서 프로필, 포스터 중심)
  - 예약 페이지 검색 기능 활성화 (인플루언서 이름/위치 기반 실시간 필터링)
  - 검색창 UI 개선 (돋보기 아이콘 우측 배치, 위치 텍스트 색상 조정)
  - 아이콘 색상 통일 (#808080) 및 버튼 스타일 일관성 개선
  - 네비게이션 바 제거에 따른 하단 버튼 위치 조정 (bottom-20 → bottom-0)

khanz0613 committed:
[fix] 라우터 설정 오류 수정 및 누락된 페이지 라우트 추가

  - 누락된 컴포넌트 import 추가 (FanMeetingPage, FanCardPage, MarketPage, MyPage)
  - 중복된 route name 수정 (FanMeetingDetailTest로 변경)
  - 팬미팅 예약 플로우 관련 라우트 추가 (좌석 선택, 결제, 성공 페이지)

khanz0613 committed:
feat: 결제 및 브랜딩 아이콘 파일들 추가

- 결제 방법 아이콘 추가 (KB페이, 카카오페이, 토스페이)
- 성공 체크 아이콘 및 포스터 이미지 추가
- 여단오 프로필 아이콘 추가
- 이미지 폴더 구조 정리

khanz0613 committed:
feat: 공통 컴포넌트들 (ButtonNav, PaymentMethodSelector) 추가

- BottomButton 컴포넌트로 하단 고정 버튼 UI 통일
- PaymentMethodSelector 컴포넌트로 결제 방법 선택 기능 구현
- 재사용 가능한 컴포넌트 구조로 설계

khanz0613 committed:
[feat] 팬미팅 데이터 구조 파일 추가

- 팬미팅 정보 중앙화된 데이터 관리
- 인플루언서 목록 및 상세 정보 구조화
- 확장 가능한 데이터 모델 설계

khanz0613 committed:
[feat] 예약 시스템 라우터 설정 추가

- 팬미팅 예약 플로우 라우트 구성
- 동적 라우팅으로 팬미팅 ID 처리
- 결제 완료 페이지 라우트 통합

khanz0613 committed:
[feat] 앱 네비게이션 로직 및 스크롤 초기화 기능 구현

- 특정 페이지에서 하단 네비게이션 숨김 처리
- 페이지 전환 시 스크롤 위치 자동 초기화
- 사용자 경험 개선을 위한 네비게이션 로직 추가

khanz0613 committed:
[feat] AppHeader 컴포넌트 뒤로가기 기능 및 스타일 개선

- 뒤로가기 버튼 클릭 이벤트 핸들러 추가
- 헤더 레이아웃 및 스타일 개선

khanz0613 committed:
[feat] AppHeader 컴포넌트 뒤로가기 기능 및 스타일 개선

- 뒤로가기 버튼 클릭 이벤트 핸들러 추가
- 헤더 레이아웃 및 스타일 개선

khanz0613 committed:
[feat] SearchBar 컴포넌트 스타일 정리 및 노출 기능 추가

- Tailwind CSS 클래스 정리 및 일관성 개선
- 검색어 키워드 외부 노출 기능 추가 (defineExpose)
- 코드 포매팅 및 가독성 향상

khanz0613 committed:
[feat] 팬미팅 목록 페이지 검색 기능 및 UI 개선

- 실시간 검색 기능 구현
- 검색 결과 없음 상태 처리
- 여단오 프로필 이미지 적용
- 데이터 구조 개선 및 외부 데이터 파일 연동

khanz0613 committed:
[feat] 팬미팅 목록 페이지 검색 기능 및 UI 개선

- 실시간 검색 기능 구현
- 검색 결과 없음 상태 처리
- 여단오 프로필 이미지 적용
- 데이터 구조 개선 및 외부 데이터 파일 연동

khanz0613 committed:
[feat] 팬미팅 상세 페이지 레이아웃 및 데이터 표시 개선

- 헤더 컴포넌트 통합 및 일관된 네비게이션
- 포스터 이미지 동적 표시 (여단오 전용 포스터)
- BottomButton 컴포넌트 적용
- 라우팅 경로 개선

khanz0613 committed:
[feat] 팬미팅 상세 페이지 레이아웃 및 데이터 표시 개선

- 헤더 컴포넌트 통합 및 일관된 네비게이션
- 포스터 이미지 동적 표시 (여단오 전용 포스터)
- BottomButton 컴포넌트 적용
- 라우팅 경로 개선

khanz0613 committed:
[feat] 좌석 선택 페이지 UI/UX 개선

- 좌석 선택 인터페이스 개선
- 사용자 친화적인 좌석 표시
- 선택된 좌석 정보 표시 개선

khanz0613 committed:
[feat] 좌석 선택 페이지 UI/UX 개선

- 좌석 선택 인터페이스 개선
- 사용자 친화적인 좌석 표시
- 선택된 좌석 정보 표시 개선

khanz0613 committed:
[feat] 결제 페이지 전면 리뉴얼 및 결제 방법 선택 기능 구현

- 전체 레이아웃 리디자인 (회색 배경, 카드형 UI)
- PaymentMethodSelector 컴포넌트 통합
- 주문 정보 표시 개선 (수수료 분리 표시)
- 동적 팬미팅 데이터 연동
- BottomButton 컴포넌트 적용

khanz0613 committed:
[feat] 결제 페이지 전면 리뉴얼 및 결제 방법 선택 기능 구현

- 전체 레이아웃 리디자인 (회색 배경, 카드형 UI)
- PaymentMethodSelector 컴포넌트 통합
- 주문 정보 표시 개선 (수수료 분리 표시)
- 동적 팬미팅 데이터 연동
- BottomButton 컴포넌트 적용

khanz0613 committed:
[refactor] 불필요한 PaymentCompletePage 파일 제거

- 추후 깃에 올라온 페이지를 받아와 적용 예정

khanz0613 committed:
소프트 리셋 할 거

khanz0613 committed:
[Refactor]소프트 리셋 진행

This reverts commit 033dd8bf9a451a129a1ab8c96528e2a6f1f47f92.

khanz0613 committed:
[feat] 예약 페이지 구조 개선 및 리베이스 오류 수정

  - 예약 관련 페이지들을 src/pages/reservation/ 폴더로 이동
  - AppHeader 중복 import 및 파싱 오류 수정
  - 라우터 import 경로 수정 및 중복 제거
  - 사용하지 않는 변수 및 함수 정리
  - 헤더와 콘텐츠 간격 통일 (20px)

khanz0613 committed:
[fix] 예약 페이지 헤더 여백 수정

 - 검색창이 헤더에 가려지는 문제 해결

khanz0613 committed:
[refactor] 예약 페이지 디자인 시스템 가이드라인 적용

- 하드코딩된 컬러를 커스텀 Tailwind 클래스로 변경
- 인라인 스타일 제거 및 표준 클래스 적용
- 컨테이너 사이즈 가이드라인 준수
- 코드 일관성 및 유지보수성 향상

khanz0613 committed:
[fix] 컨테이너 사이즈 및 레이아웃 오류 수정

좌우 여백이 맞지 않아 추가로 조정.

icegosimperson commented:
고생 많으셨습니다 👍

AshtonSW commented on src/App.vue:
각 페이지 컴포넌트에서 <AppNav />를 직접 넣고 빼기로 결정해서 전역에서 <AppNav v-if="shouldShowNav" /> 조건부 렌더링하는 로직은 없어도 될 것 같습니다! shouldShowNav 계산 로직도요!

AshtonSW replied:
스크롤 맨위로 이동하는 건 남겨도 될 것 같은데... 이건 추후에 제가 한번 확인하겠습니다!

khanz0613 replied:
> 각 페이지 컴포넌트에서 를 직접 넣고 빼기로 결정해서 전역에서 조건부 렌더링하는 로직은 없어도 될 것 같습니다! shouldShowNav 계산 로직도요!

저도 처음에 뺐었는데, 홈 페이지에 네이게이션이 없어서 혹시 다른 아직 작업 안 된 페이지에도 필요한 곳이 있을까봐 유지했었습니다!

shouldShowNav 제거하고, 홈 페이지에 그냥 하나 넣는 식으로 바꾸겠습니다~

AshtonSW replied:
👍

AshtonSW commented on src/components/common/ButtonNav.vue:
```diff
@@ -0,0 +1,37 @@
+<template>
+  <div class="fixed bottom-0 left-0 right-0 bg-white border-t border-gray-200 z-50" style="border-top-left-radius: 8px; border-top-right-radius: 8px;">
```
불필요한 style 속성이 있습니다!
테일윈드에서 class="rounded-t-2"로 대체 가능합니다!
코드 통일성을 위해 수정 부탁드립니다!

AshtonSW commented on src/pages/reservation/FanMeetingDetailPage.vue:
```diff
@@ -0,0 +1,166 @@
+}
+</script>
+
+<style scoped>
```
이건 삭제해도 될 것 같아요!

AshtonSW commented on src/pages/reservation/FanMeetingPage.vue:
```diff
@@ -0,0 +1,99 @@
+  if (!influencerList.value || !Array.isArray(influencerList.value)) {
+    return []
+  }
+  const searchKeyword = searchBarRef.value?.keyword || ''
```
이 부분은 반응형 문제가 있을 것 같은데..! 제가 한번 나중에 체크하겠습니다!

그때 검색 기능이 잘 되는 것으로 저도 같이 확인했는데, Vue 반응형 시스템에서는 .value까지 접근해야 변화 감지가 정확하게 작동한다고 합니다. keyword가 ref로 선언되어 있어서 keyword.value로 접근하지 않으면 Vue가 추적을 못하는 경우가 생긴다고 합니다!

AshtonSW commented on src/pages/reservation/FanMeetingPaymentPage.vue:
```diff
@@ -0,0 +1,165 @@
+const route = useRoute()
+
+const selectedSeat = ref('')
+const ticketPrice = ref('33,000')
```
가격 관련 부분은 문자열이 아닌 숫자로 관리되어야 할 것 같습니다!

toLocaleString으로 포맷해서 보여주고 내부 로직은 전부 숫자로 관리하는 게 좋아보여요! 추후에 데이터 넣게 되면 수정하면 될 것 같습니다!

AshtonSW reviewed (approved):
너무 훌륭하십니다!! 든든한 경한님 고생 많으셨습니다! :D


다음에는 PR을 조금 더 잘게 나눠서 올려주시면.. 감사하겠습니다..
그래야 병합할 때도 코드리뷰도 서로 모두 편할 것 같아요ㅎㅎ

고생 많으셨습니다!!

khanz0613 commented:
> 너무 훌륭하십니다!! 든든한 경한님 고생 많으셨습니다! :D
> 
> 다음에는 PR을 조금 더 잘게 나눠서 올려주시면.. 감사하겠습니다.. 그래야 병합할 때도 코드리뷰도 서로 모두 편할 것 같아요ㅎㅎ
> 
> 고생 많으셨습니다!!

항상 든든한 코드리뷰 감사합니다 :) 확실히 좀 더 잦은 PR을 올리는 것이 맞는 것 같습니다ㅜ 앞으로 인지하며 코드 짜도록 하겠습니다!

khanz0613 committed:
[refactor] 예약 페이지 리뷰 피드백 반영

- 전역 shouldShowNav 로직 제거 및 각 페이지에서 직접 관리
- ButtonNav 인라인 스타일을 Tailwind 클래스로 변경 (rounded-t-2)
- FanMeetingDetailPage 빈 style 태그 제거
- FanMeetingPage 반응형 검색 로직 개선 (toValue 사용)
- FanMeetingPaymentPage 가격을 숫자로 관리하도록 개선

AshtonSW commented:
반영해주세요! :)
